Assets/Script/Song/SongContainer.cs

```diff
@@ -280,6 +286,32 @@ public static SongEntry GetRandomSong()
             return _songs.Pick();
         }
 
+        private static SongCategory[] GetPlaycounts()
+        {
+            var player = PlayerContainer.Players.First();
+            var counts = ScoreContainer.GetPlayedSongsForUserByPlaycount(player.Profile, true);
+            // Get all the unplayed songs and stuff them on the end of the list
+            var zeroPlaySongs = new List<SongEntry>();
+            var previousSort = SettingsManager.Settings.PreviousLibrarySort;
+
+            if (previousSort == SortAttribute.Unspecified)
+            {
+                // I don't think this should ever happen, but I'm not certain,
+                // so belt and suspenders wins.
+                previousSort = SortAttribute.Name;
+            }
+
+            foreach (var category in GetSortedCategory(previousSort))
+            {
+                var songs = category.Songs.Except(counts);
+                zeroPlaySongs.AddRange(songs);
```

Comment:
Please refrain from LINQ. We should always aim to dodge hidden allocations of enumerable containers.
```suggestion
                foreach (var song in category.Songs)
                {
                    if (!counts.Contains(song))
                    {
                        zeroPlaySongs.Add(song);
                    }
                }
```

Reply:
I get the concern in the game loop, but I do wonder why the use of LINQ is a problem in menus etc?

Reply:
You already changed it but:

Whenever you perform some LINQ operation that creates a collection, especially within a loop, you add extra pressure on GC due to the collection that's generated behind the scenes. It may not seem like it's too important due to it being a menu, but GC may hold onto that collection for a while. That creates a higher likelihood that gameplay will halt from GC collecting that memory back.